src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c

```diff
@@ -420,94 +488,102 @@ void func_80BE5818(Actor* thisx, PlayState* play2) {
 
         if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
-            if (ENTANRON5_GET(&this->actor) < ENTANRON5_108) {
-                Math_Vec3f_Copy(&sp5C, &this->actor.world.pos);
-                sp5C.y = this->actor.floorHeight;
+            if (TWINMOLD_PROP_GET_TYPE(&this->actor) <= TWINMOLD_PROP_TYPE_FRAGMENT_SMALL_1) {
+                Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+                pos.y = this->actor.floorHeight;
 
                 for (i = 0; i < 4; i++) {
-                    func_80BE4930(play->specialEffects, &sp5C, Rand_ZeroFloat(1.0f) + 2.0f);
+                    EnTanron5_SpawnEffectSand(play->specialEffects, &pos, Rand_ZeroFloat(1.0f) + 2.0f);
                 }
-                this->unk_1A0++;
+
+                // Set the sinkTimer to a non-zero value so that this fragment will start sinking on the next update.
+                this->sinkTimer++;
             } else {
+                // Small ruin fragments don't sink into the sand; they just immediately despawn.
                 Actor_Kill(&this->actor);
             }
         }
     } else {
-        sp6C = 400.0f;
-        this->unk_198 += 0x2000;
+        interactionDistSq = SQ(20.0f);
+        this->itemDropRotZ += 0x2000;
         if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
-            this->unk_198 = 0;
+            this->itemDropRotZ = 0;
             this->actor.speed = 0.0f;
         }
     }
 
-    if (this->unk_1A1 == 0) {
-        if ((D_80BE5DD0 > 0.5f) &&
-            ((ENTANRON5_GET(&this->actor) < ENTANRON5_108) || (ENTANRON5_GET(&this->actor) >= ENTANRON5_110))) {
+    if (this->hitTimer == 0) {
+        if ((sGiantModeScaleFactor > 0.5f) &&
+            ((TWINMOLD_PROP_GET_TYPE(&this->actor) <= TWINMOLD_PROP_TYPE_FRAGMENT_SMALL_1) ||
+             (TWINMOLD_PROP_GET_TYPE(&this->actor) >= TWINMOLD_PROP_TYPE_ITEM_DROP_1))) {
             Player* player = GET_PLAYER(play);
-            Vec3f temp;
+            Vec3f pos;
 
-            temp.x = player->actor.world.pos.x - this->actor.world.pos.x;
-            temp.y = (player->actor.world.pos.y + 10.0f) - this->actor.world.pos.y;
-            temp.z = player->actor.world.pos.z - this->actor.world.pos.z;
+            pos.x = player->actor.world.pos.x - this->actor.world.pos.x;
+            pos.y = (player->actor.world.pos.y + 10.0f) - this->actor.world.pos.y;
+            pos.z = player->actor.world.pos.z - this->actor.world.pos.z;
 
-            if (SQXYZ(temp) < sp6C) {
-                if (ENTANRON5_GET(&this->actor) >= ENTANRON5_110) {
-                    if (this->unk_1A0 == 0) {
+            if (SQXYZ(pos) < interactionDistSq) {
+                if (TWINMOLD_PROP_GET_TYPE(&this->actor) >= TWINMOLD_PROP_TYPE_ITEM_DROP_1) {
+                    if (this->itemDropType == TWINMOLD_PROP_ITEM_DROP_TYPE_10_ARROWS) {
                         Item_Give(play, ITEM_ARROWS_10);
                     } else {
                         Item_Give(play, ITEM_MAGIC_LARGE);
```

Comment:
```suggestion
                        Item_Give(play, ITEM_MAGIC_JAR_BIG);
```
I'm surprised no conflicts were detected, but this will be updated once master is merged. Could you update:
`TWINMOLD_PROP_ITEM_DROP_TYPE_LARGE_MAGIC` -> `TWINMOLD_PROP_ITEM_DROP_TYPE_MAGIC_JAR_BIG`
as well as the file description:
`player a large amount of magic`.

Looking at this file also makes me realize `gDropMagicLargeTex` needs updating, but I can do that separately in the next cleanup

Reply:
I merged `master` and it caused no issues. I think it's because git knows I haven't changed this line, so it automatically do the merge for me.